src/sentry/api/endpoints/prompts_activity.py

```diff
@@ -35,7 +35,7 @@ def get(self, request):
 
         # always return dismissed if we are in demo mode
         if settings.DEMO_MODE:
-            return Response({"dismissed_ts": 1})
+            return Response({"data": {"dismissed_ts": 1}})
```

Comment:
I would like this to be in the demo middleware but I couldn't find a good way of handing it there

Reply:
You can maybe override endpoints?

Reply:
@EvanPurkhiser by monkey patching or something? Not sure _how_ that would be done.

Reply:
Middleware can look at the request in either the `__call__` hook, or in `process_view()` and return a response. You could have something like 

```python
if request.path == resolve("sentry-api-0-prompts-activity"):
   return Response({"data": {"dismissed_ts": 1})
```

This could work well if there are only a few endpoints that need overrides. Consolidating a big chunk of the demo mode logic into the middleware is a benefit of this approach. The drawbacks of handling this kind of logic in the middleware is that the endpoint code is no longer the source of truth on how the endpoint will work in all scenarios.

Reply:
@markstory Yea, I like this approach. I don't think we will have more than a few routes we need to override (the next one likely to be org creation).

Reply:
Definitely move the resolve call out of the hook though, no need to resolve that every time :)